modules/tensor_mechanics/src/userobjects/CrystalPlasticityStateVariable.C

```diff
@@ -42,24 +46,58 @@ CrystalPlasticityStateVariable::CrystalPlasticityStateVariable(const InputParame
 void
 CrystalPlasticityStateVariable::initSlipSysProps(std::vector<Real> & val) const
 {
-  readFileInitSlipSysRes(val);
+  switch (_intvar_read_type)
+  {
+    case 0:
+      readInitialValueFromFile(val);
+      break;
+    default:
+      readInitialValueFromInline(val);
+  }
 }
 
 void
-CrystalPlasticityStateVariable::readFileInitSlipSysRes(std::vector<Real> & val) const
+CrystalPlasticityStateVariable::readInitialValueFromFile(std::vector<Real> & val) const
 {
-  MooseUtils::checkFileReadable(_slip_sys_res_prop_file_name);
+  MooseUtils::checkFileReadable(_state_variable_file_name);
 
   std::ifstream file;
-  file.open(_slip_sys_res_prop_file_name.c_str());
+  file.open(_state_variable_file_name.c_str());
 
   for (unsigned int i = 0; i < _variable_size; ++i)
     if (!(file >> val[i]))
-      mooseError("Error CrystalPlasticityStateVariable: Premature end of slip_sys_res_prop file");
+      mooseError("Error CrystalPlasticityStateVariable: Premature end of state_variable file");
 
   file.close();
 }
 
+void
+CrystalPlasticityStateVariable::readInitialValueFromInline(std::vector<Real> & val) const
+{
+  if (_groups.size() <= 0)
+    mooseError("CrystalPlasticityStateVariable: Error in reading initial state variable values: Specify input in .i file or in state_variable file");
+  else if (_groups.size() != (_group_values.size() + 1))
+    mooseError("CrystalPlasticityStateVariable: The size of the groups and group_values does not match.");
+
+  for (unsigned int i = 0; i < _groups.size() - 1; ++i)
+  {
+    unsigned int is, ie;
+
+    is = _groups[i];
+    ie = _groups[i + 1] - 1;
+
+   if (is > ie)
+      mooseError("CrystalPlasticityStateVariable: Start index is = " << is << " should be greater than end index ie = " << ie << " in state variable read");
+
+    for (unsigned int j = is; j <= ie; ++j)
+      val[j] = _group_values[i];
+  }
+
+  for (unsigned int i = 0; i < _variable_size; ++i)
+    if (val[i] <= 0.0)
+      mooseError("CrystalPlasticityStateVariable: Value of state variables " << i  << " non positive");
```

Comment:
why do you only check this in this function and not in the read from file function? Would it make sense to move the check to right after the switch block in the constructor?